scipy/optimize/_highs/cython/src/_highs_wrapper.pyx

```diff
@@ -663,13 +665,53 @@ def _highs_wrapper(
             'message': highs.highsModelStatusToString(model_status).decode(),
             'simplex_nit': info.simplex_iteration_count,
             'ipm_nit': info.ipm_iteration_count,
-            #'fun': info.objective_function_value,
             'fun': None,
             'crossover_nit': info.crossover_iteration_count,
         }
     # If the model status is such that the solution can be read
     else:
         # Should be safe to read the solution:
+        if highs.getRanging(highsRanging) == HighsStatusOK:
+            ranging = {
+                'col_cost_up' : {
+                    'val': [highsRanging.col_cost_up.value_[ii] for ii in range(highsRanging.col_cost_up.value_.size())],
```

Comment:
I need to re-evaluate this -- I think there's a better to copy the contents of `std::vector` into a numpy array or do it in a way that doesn't involve a copy